Sources/Features/Client/Model/AttachmentsStorage.swift

```diff
@@ -0,0 +1,129 @@
+
+final class AttachmentsStorage {
+    struct Config {
+        let cacheUrl: URL
+        let directoryUrl: URL
+        let fileUrl: URL
+        
+        init(fileName: String) throws {
+            guard let cacheDirectoryURL =
+                FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+                    throw FileError.noCacheDirectory
+            }
+            self.cacheUrl = cacheDirectoryURL
+            self.directoryUrl = cacheDirectoryURL.appendingPathComponent(directoryName)
+            self.fileUrl = directoryUrl.appendingPathComponent("\(fileName)_attachments.plist")
+        }
+    }
+    
+    private static let directoryName = Bundle(for: AttachmentsStorage.self).bundleIdentifier ?? "BacktraceCache"
+    
+    static func store(_ attachments: Attachments, fileName: String) throws {
+        let config = try Config(fileName: fileName)
+        
+        if !FileManager.default.fileExists(atPath: config.directoryUrl.path) {
+            try FileManager.default.createDirectory(atPath: config.directoryUrl.path,
+                                                    withIntermediateDirectories: false,
+                                                    attributes: nil)
+        }
+        
+        let attachmentBookmarks = try convertAttachmentUrlsToBookmarks(attachments)
+        
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            try (attachmentBookmarks as NSDictionary).write(to: config.fileUrl)
+        } else {
+            guard (attachmentBookmarks as NSDictionary).write(to: config.fileUrl, atomically: true) else {
+                throw FileError.fileNotWritten
+            }
+        }
+        BacktraceLogger.debug("Stored attachments at path: \(config.fileUrl)")
+    }
+    
+    static func retrieve(fileName: String) throws -> Attachments {
+        let config = try Config(fileName: fileName)
+        guard FileManager.default.fileExists(atPath: config.fileUrl.path) else {
+            throw FileError.fileNotExists
+        }
+        // load file to NSDictionary
+        let dictionary: NSDictionary
+        if #available(iOS 11.0, tvOS 11.0, macOS 10.13, *) {
+            dictionary = try NSDictionary(contentsOf: config.fileUrl, error: ())
+        } else {
+            guard let dictionaryFromFile = NSDictionary(contentsOf: config.fileUrl) else {
+                throw FileError.invalidPropertyList
+            }
+            dictionary = dictionaryFromFile
+        }
+        
+        guard let bookmarks = dictionary as? Bookmarks else {
+            BacktraceLogger.debug("Could not convert stored dictionary to Bookmarks type")
+            throw FileError.unsupportedScheme
+        }
+        guard let attachments = try? extractAttachmentUrls(bookmarks) else {
+            BacktraceLogger.debug("Could not extract attachment URLs from stored attachments Bookmarks")
+            throw FileError.invalidPropertyList
```

Comment:
The thrown error does not match the failure reason.

Reply:
I will create a new error type for these errors, thank you!